Source/MQTTnet/Internal/Memory.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Runtime.CompilerServices;
+
+namespace MQTTnet.Internal
+{
+    public static class Memory
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void Copy(byte[] source, int sourceIndex, byte[] destination, int destinationIndex, int length)
+        {
+#if NETCOREAPP3_1_OR_GREATER || NETSTANDARD2_1
+            source.AsSpan(sourceIndex, length).CopyTo(destination.AsSpan(destinationIndex, length));
+#elif NET461_OR_GREATER || NETSTANDARD1_3_OR_GREATER
```

Comment:
Personally I'd leave these "old" target with `Array.Copy` and give the goodness only to newer targets.
But I'm fine with this change too.

Reply:
Since .NET 4.8 is still supported and used a lot I would like to keep the optimization even for "old" targets.